bit_stream_filter_context.go

```diff
@@ -0,0 +1,81 @@
+package astiav
+
+//#cgo pkg-config: libavcodec
+//#include <libavcodec/bsf.h>
+import "C"
+import (
+	"errors"
+	"unsafe"
+)
+
+// https://github.com/FFmpeg/FFmpeg/blob/release/5.1/libavcodec/bsf.h#L68
+type BSFContext struct {
+	c *C.struct_AVBSFContext
+}
+
+func newBSFContextFromC(c *C.struct_AVBSFContext) *BSFContext {
+	if c == nil {
+		return nil
+	}
+	bsfCtx := &BSFContext{c: c}
+	classers.set(bsfCtx)
+	return bsfCtx
+}
+
+var _ Classer = (*BSFContext)(nil)
+
+func AllocBitStreamContext(f *BitStreamFilter) (*BSFContext, error) {
+	if f == nil {
+		return nil, errors.New("astiav: bit stream filter must not be nil")
+	}
+
+	var bsfCtx *C.struct_AVBSFContext
+	if err := newError(C.av_bsf_alloc(f.c, &bsfCtx)); err != nil {
+		return nil, err
+	}
+
+	return newBSFContextFromC(bsfCtx), nil
+}
+
+func (bsfCtx *BSFContext) Class() *Class {
+	return newClassFromC(unsafe.Pointer(bsfCtx.c))
+}
+
+func (bsfCtx *BSFContext) Init() error {
+	return newError(C.av_bsf_init(bsfCtx.c))
+}
+
+func (bsfCtx *BSFContext) SendPacket(p *Packet) error {
+	if p == nil {
```

Comment:
The doc states that the packet can be nil (you need this to flush the context properly). Therefore I would do something like this instead:
```go
var pc *C.struct_AVPacket
if p != nil {
  pc = p.c
}
return newError(C.av_bsf_send_packet(bsfc.c, pc))
```